lib/charms/opensearch/v0/opensearch_profile.py

```diff
@@ -0,0 +1,293 @@
+        return _1GB_IN_KB
+
+    def __hash__(self):
+        """Get the hash of the profile."""
+        return hash(self.type)
+
+    def __eq__(self, value: object) -> bool:
+        """Check equality with another OpenSearchProfile."""
+        return self.type == value.type if isinstance(value, OpenSearchProfile) else False
+
+
+class ProductionProfile(OpenSearchProfile):
+    """Production profile for opensearch.
+
+    Ensures cluster meets production minimal requirements
+    """
+
+    type = PerformanceType.PRODUCTION
+
+    @property
+    def memory_requirements(self) -> ProfileMemoryRequirements:
+        """Get the memory requirements for this profile."""
+        return ProfileMemoryRequirements(
+            memory_size=8 * _1GB_IN_KB,
+            jvm_heap_percentage=0.5,
+        )
+
+    @property
+    def cluster_topology_requirements(self) -> ClusterTopologyRequirements:
+        """Get the cluster topology requirements for this profile."""
+        return ClusterTopologyRequirements(
+            cluster_managers=3,
+            data=3,
+        )
+
+
+class TestingProfile(OpenSearchProfile):
+    """Testing profile for opensearch.
+
+    Ensures basic system requirements and 1 CM+ 1 Data roles.
+    """
+
+    type = PerformanceType.TESTING
+
+    @property
+    def memory_requirements(self) -> ProfileMemoryRequirements:
+        """Get the memory requirements for this profile."""
+        return ProfileMemoryRequirements(
+            memory_size=None,
+            jvm_heap_percentage=None,
+        )
+
+    @property
+    def cluster_topology_requirements(self) -> ClusterTopologyRequirements:
+        """Get the cluster topology requirements for this profile."""
+        return ClusterTopologyRequirements(
+            cluster_managers=1,
+            data=1,
+        )
+
+
+class ProfilesManager:
+    """Manage all profile related operations"""
+
+    def __init__(self, state: "OpenSearchClusterState", workload: "OpenSearchDistribution"):
+        self.state = state
+        self.workload = workload
+        if self.profile.type == PerformanceType.TESTING:
+            logger.warning(
+                "Testing profile is used. This profile is not suitable for production use and should only be used for testing purposes."
+            )
+
+    def _apply_system_requirement(self, system_requirement: str, value: int) -> bool:
+        """Apply a system requirement."""
+        try:
+            self.workload._run_cmd(f"sysctl -w {system_requirement}={value}")
+            return int(self.workload._run_cmd(f"sysctl -n {system_requirement}")) == value
+        except OpenSearchCmdError:
+            return False
+
+    def _get_kernel_property_value(self, prop: str) -> int:
+        """Get the value of a kernel parameter."""
+        return int(self.workload._run_cmd(f"sysctl -n {prop}"))
+
+    def check_missing_system_requirements(self) -> List[str]:
```

Comment:
A note on these, given the length of the potential missing system requirements exceeding the juju status 120 chars limit, we need to log this as an error (in addition to the BlockedStatus)